src/app/store/entity/events.service.ts

```diff
@@ -0,0 +1,89 @@
+import {Injectable} from '@angular/core';
+import {Subject} from 'rxjs/Subject';
+import {ConfigService} from '../../config.service';
+import { Restangular } from 'ng2-restangular';
+
+export class ChangeEvent {
+  action: string;
+  kind: string;
+  id: string;
+}
+class MessageEvent {
+  id: string;
+  data: string;
+  event: string;
+}
+
+@Injectable()
+export class EventsService {
+
+  private eventSource: EventSource;
+  private webSocket: WebSocket;
+
+  messageEvents: Subject<String> = new Subject<String>();
+  changeEvents: Subject<ChangeEvent> = new Subject<ChangeEvent>();
+
+  constructor(config: ConfigService, restangular: Restangular) {
+    // If we have trouble setting up the EventSource
+    if (!this.connectEventSource( restangular, config.getSettings().apiEndpoint )) {
+      // , fallback to using a WebSocket
+      let wsurl = config.getSettings().apiEndpoint;
+      // I know hacky!  need to figure out how to put the wsevents resource under the apiEndpoint
+      wsurl = wsurl.replace( /^http/, 'ws' ).replace( /\/api\/v\d+$/, '/wsevents' );
```

Comment:
Yep that's what I initially tried but the websocket handler would not trigger if not at a top level path. Not sure why yet.  Perhaps if I use a different WS handling strategy it might work better (I'm currently using the SB abstractions).